compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -11358,17 +11358,25 @@ private STNode parseTupleMemberTypeDescList() {
         }
 
         // Parse first typedesc, that has no leading comma
+        STNode annot = parseOptionalAnnotations();
         STNode typeDesc = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_TUPLE);
-
-        return parseTupleTypeMembers(typeDesc, typeDescList);
+        return parseTupleTypeMembers(annot, typeDesc, typeDescList);
     }
 
     private STNode parseTupleTypeMembers(STNode typeDesc, List<STNode> typeDescList) {
-        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
+        return parseTupleTypeMembers(STNodeFactory.createEmptyNode(), typeDesc, typeDescList);
+    }
 
+    private STNode parseTupleTypeMembers(STNode annot, STNode typeDesc, List<STNode> typeDescList) {
+        typeDesc = parseComplexTypeDescriptor(typeDesc, ParserRuleContext.TYPE_DESC_IN_TUPLE, false);
         STNode tupleMemberRhs = parseTypeDescInTupleRhs();
         if (tupleMemberRhs != null) {
+            if (annot != null) {
+                reportInvalidMetaData(annot, "tuple rest descriptor");
```

Comment:
underline `reportInvalidMetaData()`, we call `addInvalidNodeToNextToken()`. What it does is the invalid token is attached as the leading minutiae for the immediate token we `consume()`. Here invalid annotation has to be the leading minutiea of the `typeDesc` node which has been already consumed. 

Therefore, correct thing to do here is (otherwise invalid token will be on a wrong place in the tree messing up the node position calculation),
`typeDesc = SyntaxErrors.cloneWithLeadingInvalidNodeMinutiae(typeDesc, annot, DiagnosticErrorCode.ERROR_INVALID_TOKEN);`

Maybe we can use define a new diagnostic code as `metadata` is referred to both annotations + documentation.
We can specifically say "Annotations are not allowed for tuple rest descriptor".